src/main/java/uk/gov/companieshouse/officerfiling/api/validation/OfficerUpdateValidator.java

```diff
@@ -166,6 +170,26 @@ public void validateOccupationSection(HttpServletRequest request, List<ApiError>
         validateOccupation(request, errorList, dto);
     }
 
+    public void validateCorrespondenceAddressSection(HttpServletRequest request, List<ApiError> errorList, OfficerFilingDto dto, AppointmentFullRecordAPI appointment) {
+        // If hasBeenUpdated boolean is true or null then continue
+        if (Boolean.FALSE.equals(dto.getCorrespondenceAddressHasBeenUpdated())) {
+            return;
+        }
+        // If address isn't null and any field within this section has been provided then continue
+        final AddressDto address = dto.getServiceAddress();
+        final boolean addressIsNull = address == null || (address.getPremises() == null && address.getAddressLine1() == null && address.getAddressLine2() == null && address.getLocality() == null && address.getRegion() == null && address.getPostalCode() == null && address.getCountry() == null);
```

Comment:
Should we check for StringUtils.isBlank()? Maybe this could be a private method as rather a long line and may be missed if address fields are changed or added to.